eng/pipelines/pr-breaking-check.yml

```diff
@@ -0,0 +1,94 @@
+          - checkout: self
+            submodules: true
+          - template: /eng/pipelines/templates/install.yml
+            parameters:
+              nodeVersion: "18.x"
+              workingDirectory: $(Build.SourcesDirectory)/typespec-azure
+          - script: pnpm run build
+            displayName: Build
+            workingDirectory: $(Build.SourcesDirectory)/typespec-azure
+
+          #Step 2: Update swagger by TypeSpec in Step 1
+          - checkout: azure-rest-api-specs
+            fetchDepth: 1
+          - pwsh: |
+              $packageJson = Get-Content -Raw "package.json" | ConvertFrom-Json -AsHashtable
+
+              $packageJson.devDependencies["@azure-tools/typespec-autorest"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-autorest"
+              $packageJson.devDependencies["@azure-tools/typespec-azure-core"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-azure-core"
+              $packageJson.devDependencies["@azure-tools/typespec-azure-portal-core"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-azure-portal-core"
+              $packageJson.devDependencies["@azure-tools/typespec-azure-resource-manager"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-azure-resource-manager"
+              $packageJson.devDependencies["@azure-tools/typespec-client-generator-core"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-client-generator-core"
+              $packageJson.devDependencies["@azure-tools/typespec-azure-rulesets"] = "file:$(Build.SourcesDirectory)/typespec-azure/packages/typespec-azure-rulesets"
+              $packageJson.devDependencies["@typespec/compiler"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/compiler"
+              $packageJson.devDependencies["@typespec/http"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/http"
+              $packageJson.devDependencies["@typespec/openapi"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/openapi"
+              $packageJson.devDependencies["@typespec/openapi3"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/openapi3"
+              $packageJson.devDependencies["@typespec/rest"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/rest"
+              $packageJson.devDependencies["@typespec/versioning"] = "file:$(Build.SourcesDirectory)/typespec-azure/core/packages/versioning"
+
+              $packageJson | ConvertTo-Json -Depth 100 | Out-File -Path "package.json" -Encoding utf8 -NoNewline -Force
+            displayName: "Update package.json in azure-rest-api-specs"
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+          - script: npm install
+            displayName: Install updated TypeSpec dependencies
+            workingDirectory: $(Build.SourcesDirectory)/azure-rest-api-specs
+          - pwsh: ./eng/scripts/TypeSpec-Generate-Swagger.ps1 -CheckAll
```

Comment:
Where is this script comming from I don't see it in azure-rest-api-specs.

`TypeSpec-Validation.ps1` is the one that does the regen and checks

Reply:
Yeah I put details in the description that it comes from this [PR](https://github.com/Azure/azure-rest-api-specs/pull/29646). 

`TypeSpec-Validation.ps1` would throw error if there is any change, which is not we are expecting. We hope to get the diff.